forge-gui/res/cardsfolder/upcoming/break_out.txt

```diff
@@ -0,0 +1,11 @@
+Name:Break Out
+ManaCost:R G
+Types:Sorcery
+A:SP$ PeekAndReveal | Defined$ You | PeekAmount$ 6 | NoReveal$ True | RememberPeeked$ True | SubAbility$ PickOne | StackDescription$ SpellDescription | SpellDescription$ Look at the top six cards of your library. You may reveal a creature card from among them. If that card has mana value 2 or less, you may put it onto the battlefield and it gains haste until end of turn. If you didn’t put the revealed card onto the battlefield this way, put it into your hand. Put the rest on the bottom of your library in a random order.
+SVar:PickOne:DB$ ChooseCard | Defined$ You | Amount$ 1 | ChoiceTitle$ Choose a creature card to reveal | Choices$ Card.Creature+IsRemembered | ChoiceZone$ Library | Reveal$ True | SubAbility$ DBChangeZoneBattlefield
+SVar:DBChangeZoneBattlefield:DB$ ChangeZone | Origin$ Library | Destination$ Battlefield | Defined$ ChosenCard | ConditionDefined$ Remembered | ConditionPresent$ Card.Creature+cmcLE2+ChosenCard | ConditionCompare$ EQ1 | Optional$ True | SubAbility$ DBPump | StackDescription$ None
+SVar:DBPump:DB$ Pump | KW$ Haste | Defined$ ChosenCard | SubAbility$ DBChangeZoneHand | StackDescription$ None
```

Comment:
you might also need to work with Imprint here so it only Pumps when you have it ETB

Reply:
@tool4ever maybe use AnimateSubAbility for this?

Reply:
why? it says "gains" so it shouldn't ETB with haste

Reply:
Figured out a way to implement `Imprint` on the script for the chosen creature gaining haste. I'm just not sure how to verify that the creature first enters the battlefield and then gains haste.

Reply:
well you should move the Imprint to the ChangeZone line instead
(I don't want to confuse you further but currently Pump only works on battlefield by default but ChosenCard would be the old library object which will eventually need to fail due to timestamp checks)

Reply:
Took a bit to untangle but it's done :)

Reply:
Why are you trying to Pump two things now? Imprinted will be the battlefield object

Reply:
Sorry, that was a remainder of previous digressions due to some confusion on my part. Script fixed: it functions as expected.